aws_lambda_powertools/utilities/batch/middlewares.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+
+"""
+Middlewares for batch utilities
+"""
+from typing import Callable, Dict
+
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+
+from .base import BaseProcessor
+
+
+@lambda_handler_decorator
+def batch_processor(
+    handler: Callable, event: Dict, context: Dict, record_handler: Callable, processor: BaseProcessor = None
+):
+    """
+    Middleware to handle batch event processing
+
+    Parameters
+    ----------
+    handler: Callable
+        Lambda's handler
+    event: Dict
+        Lambda's Event
+    context: Dict
```

Comment:
Theoretically it's not a dict it's has methods, I always use:

```python
from aws_lambda_context import LambdaContext

def handler(event: Dict, context: LambdaContext):
    pass
```

@heitorlessa I think it would be a good addition to provide a typing for the context parameter. 

Surely not in scope for this PR but I can write a RFC if you think it's useful?

Reply:
definitely @Nr18 - Here's a good start as it's 90% done: https://gist.github.com/alexcasalboni/a545b68ee164b165a74a20a5fee9d133

Reply:
Ah thanks, I created #144 and will have a closer look at the gist you provided later on.